tasking/pulp/tasking/storage.py

```diff
@@ -0,0 +1,123 @@
+import errno
+import os
+import shutil
+import stat
+
+from celery import task
+
+from django.conf import settings as pulp_settings
+
+
+def _working_dir_root(worker_name):
+    """
+    Returns the path to the working directory of a worker
+    :param worker_name:     Name of worker for which path is requested
+    :type  name:            basestring
+    :return working_directory setting from server config
+    :rtype  str
+    """
+    working_dir = pulp_settings.SERVER['working_directory']
+    dir_root = os.path.join(working_dir, worker_name)
+    return dir_root
+
+
+def create_worker_working_directory(worker_name):
+    """
+    Creates a working directory inside the working_directory as specified in /etc/pulp/server.yaml
+    default path for working_directory is /var/cache/pulp
+    :param worker_name:     Name of worker that uses the working directory created
+    :type  name:            basestring
+    :return Path to the working directory for a worker
+    :rtype  str
+    """
+    working_dir_root = _working_dir_root(worker_name)
+    os.mkdir(working_dir_root)
+
+
+def delete_worker_working_directory(worker_name):
+    """
+    Deletes a working directory inside the working_directory as specified in /etc/pulp/server.yaml
+    default path for working_directory is /var/cache/pulp
+    :param worker_name:     Name of worker that uses the working directory being deleted
+    :type  name:            basestring
+    """
+    working_dir_root = _working_dir_root(worker_name)
+    if os.path.exists(working_dir_root):
+        _rmtree_fix_permissions(working_dir_root)
+
+
+def _working_directory_path():
+    """
+    Gets the path for a task's working directory inside a workers working directory
+
+    @return: full path on disk to the working directory for current task
+    @rtype:  str
+    """
+    current_task = task.current
+    if (current_task and current_task.request and current_task.request.id and
+            current_task.request.hostname):
+        worker_name = current_task.request.hostname
+        task_id = current_task.request.id
+        worker_dir_root = _working_dir_root(worker_name)
+        working_dir_root = os.path.join(worker_dir_root, task_id)
+    else:
+        return None
+
+    return working_dir_root
+
+
+def delete_working_directory():
+    """
+    Deletes working directory for a particular task.
+
+    @return: None
+    @rtype:  None
+    """
+    working_dir_root = _working_directory_path()
+
+    if working_dir_root and os.path.exists(working_dir_root):
```

Comment:
Similar change requested here. Try to delete it and handle exceptions accordingly.